src/render/emitter.ts

```diff
@@ -1,9 +1,29 @@
+import { stringify, toValue } from '../util/underscore'
+
 export class Emitter {
-  public html = '';
+  public html: any = '';
   public break = false;
   public continue = false;
+  private keepOutputType? = false;
+
+  constructor (keepOutputType: boolean|undefined) {
+    this.keepOutputType = keepOutputType
+  }
 
-  public write (html: string) {
-    this.html += html
+  public write (html: any) {
+    if (this.keepOutputType === true) {
+      html = toValue(html)
+    } else {
+      html = stringify(toValue(html))
+    }
+    // This will only preserve the type if the value is isolated.
+    // I.E:
+    // {{ my-port }} -> 42
+    // {{ my-host }}:{{ my-port }} -> 'host:42'
+    if (this.keepOutputType === true && typeof html !== 'string' && this.html === '') {
+      this.html = html
+    } else {
+      this.html += html as string
```

Comment:
OK, it makes sense to be one Emitter. The previously set `this.html` can be a `toValue()`, which also needs `stringify`.

Reply:
No sure entirely sure I understand what you mean, but I hope this is what you were after. Updated in 68a5938